### PR TITLE
import account and add only if not exists

### PR DESCRIPTION
**Pull request by OrenZak**
Import account was always adding the account to the list, although it has this account.

So now it will search on the account return if the relevant account if exists.

### CODE REVIEW

File: kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java [hunk 1]
```diff
@@ -310,6 +310,30 @@ public void backupRestore() throws Exception {
         }
     }
 
+    @Test
+    public void importAccount_AddOnlyIfNotExists() throws Exception {
+        KinAccount kinAccount = kinClient.addAccount();
+        String uuid = UUID.randomUUID().toString();
```
Comment (yosriz) [inline kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: I would use the const `PASSPHRASE`, took me a while to understands what's the param (or change the var name to passphrase)

File: kin-core/src/main/java/kin/core/KinClient.java [hunk 2]
```diff
@@ -125,7 +129,20 @@ KinAccount addAccount() throws CreateAccountException {
...
-        return addKeyPair(account);
+        KinAccount kinAccount = getAccountByPublicAddress(account.getAccountId());
+        return kinAccount != null ? kinAccount : addKeyPair(account);
+    }
+
+    @Nullable
+    private KinAccount getAccountByPublicAddress(String accountId) {
+        KinAccount kinAccount = null;
+        for (int i = 0; i < kinAccounts.size(); i++) {
+            final KinAccount account = kinAccounts.get(0);
```
Comment (yosriz) [inline kin-core/src/main/java/kin/core/KinClient.java]: `0` -> `i`
add also test for multiple adding of accounts, that should be failed before fixing the `0` to `i`

**Review by yosriz — CHANGES_REQUESTED** [review]

File: kin-core/src/main/java/kin/core/KinClient.java [hunk 3]
```diff
@@ -64,8 +68,8 @@ public KinClient(@NonNull Context context, @NonNull ServiceProvider provider) {
 
     @VisibleForTesting
     KinClient(ServiceProvider serviceProvider, KeyStore keyStore, TransactionSender transactionSender,
-        AccountActivator accountActivator, AccountInfoRetriever accountInfoRetriever,
-        BlockchainEventsCreator blockchainEventsCreator, BackupRestore backupRestore) {
+              AccountActivator accountActivator, AccountInfoRetriever accountInfoRetriever,
```
Comment (yosriz) [inline kin-core/src/main/java/kin/core/KinClient.java]: redundant formatting changes, do you use code style project settings?

File: kin-core/src/main/java/kin/core/KinClient.java (hunk 2 above)
Comment (OrenZak) [reply kin-core/src/main/java/kin/core/KinClient.java]: 🥇

File: kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java (hunk 1 above)
Comment (OrenZak) [reply kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: I've copied what you did.. you don't have `PASSPHRASE`.

Comment (yosriz) [reply kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: yes, I see where it comes from, I did a loop and there I've used uuid. shame on me.
BTW, did you run those tests manually? CI can't run instrumentation tests for now.

**Review by yosriz — APPROVED** [review]

File: kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java (hunk 1 above)
Comment (OrenZak) [reply kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: yup manually.

Comment (drEfrati) [reply kin-core/src/androidTest/java/kin/core/KinClientIntegrationTest.java]: Just a little question.
After this change, will the current account be the last account in the list of accounts?
Meaning if , for example, developer will do this:
kinClient.getAccount(kinClient.accountCount - 1)
Then will he get this account?

**Comment by drEfrati** [comment]
Just a little question.
After this change, will the current account be the last account in the list of accounts?
Meaning if , for example, developer will do this:
kinClient.getAccount(kinClient.accountCount - 1)
Then will he get this account?

**Comment by OrenZak** [comment]
@drEfrati nope, the developer should get the accounts list and search for his account by address.
Maybe kinClient should have another API for: `getAccountbyAddress(publicAddress)`

**Comment by drEfrati** [comment]
@OrenZak Thanks.

**Pull request merged**